src/string/CreditCard.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * TODO: Add module comment
+ *
+ * @since 0.0.3
+ */
+import { Kind, Kind2, URIS, URIS2, HKT2 } from 'fp-ts/HKT'
+import * as D from 'io-ts/Decoder'
+import * as Enc from 'io-ts/Encoder'
+import * as Eq_ from 'fp-ts/Eq'
+import * as G from 'io-ts/Guard'
+import * as Str from 'fp-ts/string'
+import * as TD from 'io-ts/TaskDecoder'
+import * as t from 'io-ts/Type'
+import { flow, pipe } from 'fp-ts/function'
+import * as fc from 'fast-check'
+import * as RA from 'fp-ts/ReadonlyArray'
+
+import * as Arb from '../internal/ArbitraryBase'
+import { MonoidSum } from 'fp-ts/lib/number'
+import { digits } from '../internal/util'
+
+/**
+ * @since 0.0.3
+ * @category Internal
+ */
+interface CreditCardBrand {
+  readonly CreditCard: unique symbol
+}
+
+/**
+ * TODO: Add module comment
+ *
+ * @since 0.0.3
+ * @category Model
+ */
+export type CreditCard = string & CreditCardBrand
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams<S> = HKT2<S, string, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams1<S extends URIS> = Kind<S, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2<S extends URIS2> = Kind2<S, string, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2C<S extends URIS2> = Kind2<S, unknown, CreditCard>
+
+const cc =
```

Comment:
🔧 may want to add this annotation to the regex:

```ts
/**
 * @since 0.0.3
 * @category Internal
 */
```

Reply:
It's not even exported, though?

Also I'd been using `/** @internal */` for internal things because TS actually cares about that -- things tagged thus will not even exist in the emitted types.

Reply:
Good point about the `/** @internal */` I've always done the `@since / @category` just to make it homogeneous with the rest of the module